client/java-spring-boot-autoconfigure/src/main/java/com/linecorp/centraldogma/client/spring/CentralDogmaClientAutoConfiguration.java

```diff
@@ -76,15 +81,48 @@ public ClientFactory dogmaClientFactory() {
     @Bean
     public CentralDogma client(
             Environment env,
+            CentralDogmaSettings settings,
             @ForCentralDogma ClientFactory clientFactory,
             Optional<ArmeriaClientConfigurator> armeriaClientConfigurator) throws UnknownHostException {
 
-        return new LegacyCentralDogmaBuilder()
-                .clientFactory(clientFactory)
-                .profile(env.getActiveProfiles())
-                .clientConfigurator(cb -> armeriaClientConfigurator.ifPresent(
-                        configurator -> configurator.configure(cb)))
-                .build();
+        final ArmeriaCentralDogmaBuilder builder = new ArmeriaCentralDogmaBuilder();
+
+        builder.clientFactory(clientFactory);
+        builder.clientConfigurator(cb -> armeriaClientConfigurator.ifPresent(
+                configurator -> configurator.configure(cb)));
+
+        // Set health check interval.
+        final Long healthCheckIntervalMillis = settings.getHealthCheckIntervalMillis();
+        if (healthCheckIntervalMillis != null) {
+            builder.healthCheckIntervalMillis(healthCheckIntervalMillis);
+        }
+
+        // Enable or disable TLS.
+        final Boolean useTls = settings.getUseTls();
+        if (useTls != null) {
+            builder.useTls(useTls);
+        }
+
+        // Set profile or hosts.
+        final String profile = settings.getProfile();
+        final List<String> hosts = settings.getHosts();
+        if (profile != null) {
+            builder.profile(CentralDogmaClientAutoConfiguration.class.getClassLoader(), profile);
+        } else if (hosts != null) {
+            for (String h : hosts) {
+                final HostAndPort hostAndPort = HostAndPort.fromString(h);
+                if (hostAndPort.hasPort()) {
+                    builder.host(hostAndPort.getHost(), hostAndPort.getPort());
+                } else {
+                    builder.host(hostAndPort.getHost());
+                }
+            }
+        } else {
+            // Use the currently active Spring Boot profiles if neither profile nor hosts was specified.
+            builder.profile(env.getActiveProfiles());
```

Comment:
It magically works, but hard to recognize why it work...

Is this to support backward compatibility?

Reply:
Yes, and documented in `client-java.rst`.

Reply:
Let me add some logging so that a user knows where the profile name came from.